### PR TITLE
crimson/os/seastore/../segment_manager: improve logs and validations

### PR DESCRIPTION
**Pull request by cyx1231st**
formats:
* `D<device-id> S<segment-id> offset=<off>~<len> poffset=<off> information`
* `D<device-id> poffset=<off>~<len> information`

levels:
* INFO:  major initiation, closing and segment operations
* DEBUG: INFO details, major read and write operations
* TRACE: DEBUG details

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by cyx1231st** [comment]
jenkins test docs

**Review by athanatos — APPROVED** [review]

**Comment by athanatos** [comment]
retest this please

**Review by liu-chunmei — APPROVED** [review]

**Comment by cyx1231st** [comment]
jenkins retest this please

**Comment by athanatos** [comment]
retest this please

**Comment by cyx1231st** [comment]
```
215/258 Test #233: unittest-interruptible-future .............Child aborted***Exception:   0.89 sec
WARNING: debug mode. Not for benchmarking or production
WARN  2022-01-10 21:54:27,280 [shard 0] seastar - Creation of perf_event based stall detector failed, falling back to posix timer: std::system_error (error system:13, perf_event_open() failed: Permission denied)
unittest-interruptible-future: ../src/test/crimson/seastar_runner.h:45: int SeastarRunner::init(int, char **): Assertion `begin_signaled == true' failed.
Aborting.
Backtrace:
  0xb5a29a
  0x1e22b40
  0x1e2276d
  0x1c7bba2
  0x1ca7605
  0x1d7f859
  0x1d7faa1
  0x1d7f8ca
  0x7f6f0055c3bf
  /lib/x86_64-linux-gnu/libc.so.6+0x4618a
  /lib/x86_64-linux-gnu/libc.so.6+0x25858
  /lib/x86_64-linux-gnu/libc.so.6+0x25728
  /lib/x86_64-linux-gnu/libc.so.6+0x36f35
  0xe66828
  0xe59913
  /lib/x86_64-linux-gnu/libc.so.6+0x270b2
  0xb2327d
```
Not sure why only `unittest-interruptible-future` is affected by `Assertion `begin_signaled == true' failed.`. Working on a fix now.

**Review by xxhdx1985126 — APPROVED** [review]

**Comment by cyx1231st** [comment]
jenkins test make check

File: src/crimson/os/seastore/segment_manager.cc [hunk 1]
```diff
@@ -22,6 +22,66 @@ seastar::logger &logger(){
...
+  for (const auto& [k, v] : sb.secondary_devices) {
+    out << std::to_string(k) << ": " << v;
+  }
+  return out << "))";
+}
+
+std::ostream& operator<<(std::ostream& out, const segment_manager_config_t& conf)
+{
+  out << "sm_config_t("
+      << ", major_dev=" << conf.major_dev
```
Comment (cyx1231st) [inline src/crimson/os/seastore/segment_manager.cc:58]: typo

**Pull request merged**